front/pages/api/w/[wId]/subscriptions/index.ts

```diff
@@ -65,6 +79,92 @@ async function handler(
         });
       }
       break;
+    }
+
+    case "PATCH": {
+      const bodyValidation = PatchSubscriptionRequestBody.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${pathError}`,
+          },
+        });
+      }
+      const subscription = auth.subscription();
+      if (!subscription) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "subscription_not_found",
+            message: "The subscription was not found.",
+          },
+        });
+      }
+
+      const { action } = bodyValidation.right;
+
+      switch (action) {
+        case "cancel_free_trial":
+          if (!subscription.trialing) {
```

Comment:
NIT: since we're doing the same two checks for both cases maybe we can group them before the switch?